doc/user/content/overview/architecture.md

```diff
@@ -1,31 +1,99 @@
+
+### Cluster replicas
+
+Cluster replicas are the physical incarnations of a given cluster. A cluster replica represents a physical set of computational resources (CPU and RAM) that is responsible for maintaining the indexes and materialized views defined on the parent cluster.
+
+Each cluster replica associated with a given cluster contains all of the indexes and materialized views associated with that cluster. However, different cluster replicas can be differently sized, even if they are associated with the same cluster. Each replica is guaranteed to produce the same outputs as all of its peers for all times, because all cluster replicas read the same input data collections from the storage layer, and they all perform the same deterministic computations.
+
+### Indexes
+
+Indexes assemble and maintain source, view or materialized view output data in-memory on all cluster replicas associated with a given cluster. Creating an index enables future `SELECT` queries against the given source, view or materialized view to return rapidly because the requested data can be read without interacting with persistent storage, or incurring (much) additional computation.
+
+Users can create indexes on any source, unmaterialized view, or materialized view present in the region. However indexes are associated with a particular cluster, and when an index is created, altered, or destroyed, the corresponding change is replicated on all current and future replicas associated with that cluster. This setup lets users decouple changes to the computation (which views are defined, which indexes are created) from how that computation is physically deployed (how many replicas, how large is each replica).
+
+Note however, that unlike (materialized) views, indexes are local to a cluster and can can only be used on `SELECT` queries on the same cluster.
+
+For a deeper dive into how indexes work see [here](/ops/optimization/#indexes) and [here](/sql/create-index).
+
+### Materialized views
+
+Materialized views compute and then maintain views as data collections stored in persistent storage (S3). Subsequent `SELECT` queries can then read results from the stored data collection without having to perform (much) additional computation.
+
+If `SELECT` queries against a materialized view are too slow, it is recommended to create an index based on common query patterns.
+
+Materialized views are associated with a particular cluster but they can be read from any cluster in the region.
 
 ### Adapter
 
+Adapter is responsible for handling users input SQL statements, and assigning timestamps to users read (`SELECT`, `SUBSCRIBE`, `COPY ... FROM`) queries and write (`INSERT`, `UPDATE`, `DELETE`, `COPY ... TO`) queries to ensure that they receive outputs corresponding to those timestamps. The Adapter is also responsible for ensuring that [transactions](/overview/isolation-level) work correctly.
```

Comment:
should be `users'` not `users` throughout.

You could also mention that the adapter is responsible for doing bookkeeping on the frontiers of sources/dataflows which is what enables it to actually pick such a timestamp.